# **Super Store Data Analysis**

**Author Name:** ***Ahsanullah ghani***

**Email:** ***ahsanullahghani000@gmail.com***


 **The data was Collected from the following**  [[Link]](https://www.kaggle.com/datasets/vivek468/superstore-dataset-final)

# **About Dataset**

## Context

With growing demands and cut-throat competition in the market, a Superstore Giant is seeking knowledge to understand what works best for them.

They want to understand which:

- **Products**
- **Regions**
- **Categories**
- **Customer Segments**

they should **target or avoid**.

You can even take this a step further and try to build a **Regression Model** to predict **Sales** or **Profit**.

> **Go crazy with the dataset, but also make sure to provide some business insights to improve.**

---

# Metadata

| Column | Description |
|---|---|
| **Row ID** | Unique ID for each row. |
| **Order ID** | Unique Order ID for each Customer. |
| **Order Date** | Order Date of the product. |
| **Ship Date** | Shipping Date of the Product. |
| **Ship Mode** | Shipping Mode specified by the Customer. |
| **Customer ID** | Unique ID to identify each Customer. |
| **Customer Name** | Name of the Customer. |
| **Segment** | The segment where the Customer belongs. |
| **Country** | Country of residence of the Customer. |
| **City** | City of residence of the Customer. |
| **State** | State of residence of the Customer. |
| **Postal Code** | Postal Code of every Customer. |
| **Region** | Region where the Customer belongs. |
| **Product ID** | Unique ID of the Product. |
| **Category** | Category of the product ordered. |
| **Sub-Category** | Sub-Category of the product ordered. |
| **Product Name** | Name of the Product. |
| **Sales** | Sales of the Product. |
| **Quantity** | Quantity of the Product. |
| **Discount** | Discount provided. |
| **Profit** | Profit/Loss incurred. |

---

# Acknowledgements

I do not own this data. I merely found it from the **Tableau** website.

All credits to the original authors/creators.

**For educational purposes only.**

# **TASK**

# **1** **Data Loading**

***Import pandas***

In [2]:
import pandas as pd



***Load dataset***

In [5]:
df = pd.read_csv(r"C:\Users\AHSAN GHANI\Documents\6Month Intership program\Frist Month\pandas_tip&trick\Data set\store dataset\Sample - Superstore.csv", encoding="latin1")

***Display first 5  Rows***

In [6]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


***Check Dataset Shape and Columns***

In [9]:
# Check the shape
df.shape
#Check the columns
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='str')

# **2** **Data Cleaning**

***Check the Missing Values in Datase***

In [10]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

 ***Handle The Missing Values***

In [ ]:
df['Sales'] = df['Sales'].fillna(df['Sales'].mean()) 
# this command can change the missing values in the mean of data 

***Check the Duplicate***

In [14]:
df.duplicated().sum()

np.int64(0)

***Handel the duplicated***

In [ ]:
df = df.drop_duplicates()


In [17]:
df.duplicated().sum()

np.int64(0)

***Convert order data and Ship data into Datatime***

In [18]:
# first check the order date
df[['Order Date','Ship Date']]

,Order Date,Ship Date
0,11/8/2016,11/11/2016
1,11/8/2016,11/11/2016
2,6/12/2016,6/16/2016
3,10/11/2015,10/18/2015
4,10/11/2015,10/18/2015
...,...,...
9989,1/21/2014,1/23/2014
9990,2/26/2017,3/3/2017
9991,2/26/2017,3/3/2017
9992,2/26/2017,3/3/2017


In [ ]:
 # now we convert the datetime
df[['Order Date','Ship Date']] = df[['Order Date','Ship Date']].apply(pd.to_datetime)


In [22]:
df[['Order Date','Ship Date']].dtypes

Order Date    datetime64[us]
Ship Date     datetime64[us]
dtype: object

# **3** **EDA** 

***Analyze Sales ,Profits, Quantity***

In [27]:
import pandas as pd
pd.set_option('display.float_format', '{:.2f}'.format)

In [28]:
df[['Sales', 'Profit', 'Quantity']].describe()

,Sales,Profit,Quantity
count,9994.00,9994.00,9994.00
mean,229.86,28.66,3.79
std,623.25,234.26,2.23
min,0.44,-6599.98,1.00
25%,17.28,1.73,2.00
50%,54.49,8.67,3.00
75%,209.94,29.36,5.00
max,22638.48,8399.98,14.00


In [29]:
df[['Sales', 'Profit', 'Quantity']].sum(),round(2)

(Sales      2297200.86
 Profit      286397.02
 Quantity     37873.00
 dtype: float64,
 2)

 ***Group by Catagory , Sub_Catagory ,Rigion, Segment***

In [30]:


category_group = df.groupby('Category')[['Sales','Profit','Quantity']].sum()
print(category_group)

                    Sales    Profit  Quantity
Category                                     
Furniture       741999.80  18451.27      8028
Office Supplies 719047.03 122490.80     22906
Technology      836154.03 145454.95      6939


In [31]:

# Sub-Category wise profit
sub_profit = df.groupby('Sub-Category')['Profit'].sum()
sub_profit = sub_profit.sort_values()

print("Loss making sub-categories:")
print(sub_profit.head(3))

print("\nTop profitable sub-categories:")
print(sub_profit.tail(3))

Loss making sub-categories:
Sub-Category
Tables      -17725.48
Bookcases    -3472.56
Supplies     -1189.10
Name: Profit, dtype: float64

Top profitable sub-categories:
Sub-Category
Accessories   41936.64
Phones        44515.73
Copiers       55617.82
Name: Profit, dtype: float64


# **4** **Feature Engineering**

***Create Profit Ratio = Profit/ Sale***

In [33]:
# New columns: Profit Ratio
df['Profit Ratio'] = df['Profit'] / df['Sales']

print(df[['Sales', 'Profit', 'Profit Ratio']].head(30))

     Sales   Profit  Profit Ratio
0   261.96    41.91          0.16
1   731.94   219.58          0.30
2    14.62     6.87          0.47
3   957.58  -383.03         -0.40
4    22.37     2.52          0.11
5    48.86    14.17          0.29
6     7.28     1.97          0.27
7   907.15    90.72          0.10
8    18.50     5.78          0.31
9   114.90    34.47          0.30
10 1706.18    85.31          0.05
11  911.42    68.36          0.08
12   15.55     5.44          0.35
13  407.98   132.59          0.32
14   68.81  -123.86         -1.80
15    2.54    -3.82         -1.50
16  665.88    13.32          0.02
17   55.50     9.99          0.18
18    8.56     2.48          0.29
19  213.48    16.01          0.07
20   22.72     7.38          0.33
21   19.46     5.06          0.26
22   60.34    15.69          0.26
23   71.37    -1.02         -0.01
24 1044.63   240.26          0.23
25   11.65     4.22          0.36
26   90.57    11.77          0.13
27 3083.43 -1665.05         -0.54
28    9.62    

***Extract Year and Month From Order Dates***

In [35]:

#conver order data to date format
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Order Year'] = df['Order Date'].dt.year
df['Order Month'] = df['Order Date'].dt.month
print(df[['Order Date', 'Order Year', 'Order Month']])

     Order Date  Order Year  Order Month
0    2016-11-08        2016           11
1    2016-11-08        2016           11
2    2016-06-12        2016            6
3    2015-10-11        2015           10
4    2015-10-11        2015           10
...         ...         ...          ...
9989 2014-01-21        2014            1
9990 2017-02-26        2017            2
9991 2017-02-26        2017            2
9992 2017-02-26        2017            2
9993 2017-05-04        2017            5

[9994 rows x 3 columns]


***Identify Top performing and loaa making area***

In [37]:
# Group by Category and sum the Profit for each category
category_profit = df.groupby('Category')['Profit'].sum()

# Sort values from lowest to highest
category_profit = category_profit.sort_values()

print("Category wise Profit:")
print(category_profit)

# Group by Sub-Category and sum the Profit for each sub-category
subcat_profit = df.groupby('Sub-Category')['Profit'].sum()

# Sort values from lowest to highest
subcat_profit = subcat_profit.sort_values()

print("\nTop 5 Performing Sub-Categories:")
print(subcat_profit.tail(5))   # last 5 = highest profit

print("\nLoss Making Sub-Categories:")
print(subcat_profit.head(5))   # first 5 = lowest / negative profit

Category wise Profit:
Category
Furniture          18451.27
Office Supplies   122490.80
Technology        145454.95
Name: Profit, dtype: float64

Top 5 Performing Sub-Categories:
Sub-Category
Binders       30221.76
Paper         34053.57
Accessories   41936.64
Phones        44515.73
Copiers       55617.82
Name: Profit, dtype: float64

Loss Making Sub-Categories:
Sub-Category
Tables      -17725.48
Bookcases    -3472.56
Supplies     -1189.10
Fasteners      949.52
Machines      3384.76
Name: Profit, dtype: float64


# **5** **Advance Analysis**

***Identify Loss Making Product***

In [38]:

# Group by Product Name and sum the Profit for each product
product_profit = df.groupby('Product Name')['Profit'].sum()

# Sort values from lowest to highest
product_profit = product_profit.sort_values()

# Filter only products with negative profit (loss making)
loss_products = product_profit[product_profit < 0]

print("Total Loss Making Products:", len(loss_products))
print("\nTop 10 Loss Making Products:")
print(loss_products.head(10))

Total Loss Making Products: 301

Top 10 Loss Making Products:
Product Name
Cubify CubeX 3D Printer Double Head Print                           -8879.97
Lexmark MX611dhe Monochrome Laser Printer                           -4589.97
Cubify CubeX 3D Printer Triple Head Print                           -3839.99
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases            -2876.12
Bush Advantage Collection Racetrack Conference Table                -1934.40
GBC DocuBind P400 Electric Binding System                           -1878.17
Cisco TelePresence System EX90 Videoconferencing Unit               -1811.08
Martin Yale Chadless Opener Electric Letter Opener                  -1299.18
Balt Solid Wood Round Tables                                        -1201.06
BoxOffice By Design Rectangular and Half-Moon Meeting Room Tables   -1148.44
Name: Profit, dtype: float64


***Region Wise Proftability Analysis***

In [39]:
# Group by Region and calculate total Sales, Profit and Quantity
region_analysis = df.groupby('Region')[['Sales', 'Profit', 'Quantity']].sum()

# Calculate Profit Ratio (Profit / Sales) for each region
region_analysis['Profit Ratio'] = region_analysis['Profit'] / region_analysis['Sales']

# Sort by Profit from highest to lowest
region_analysis = region_analysis.sort_values('Profit', ascending=False)

print(region_analysis)

            Sales    Profit  Quantity  Profit Ratio
Region                                             
West    725457.82 108418.45     12266          0.15
East    678781.24  91522.78     10618          0.13
South   391721.91  46749.43      6209          0.12
Central 501239.89  39706.36      8780          0.08


***Montly And Yearly Trend***

In [40]:
# Convert Order Date to proper date format
df['Order Date'] = pd.to_datetime(df['Order Date'])

# Extract Year and Month
df['Order Year'] = df['Order Date'].dt.year
df['Order Month'] = df['Order Date'].dt.month

# Yearly Trend
yearly_trend = df.groupby('Order Year')[['Sales', 'Profit']].sum()
print("Yearly Trend:")
print(yearly_trend)

# Monthly Trend (combined across all years)
monthly_trend = df.groupby('Order Month')[['Sales', 'Profit']].sum()
print("\nMonthly Trend:")
print(monthly_trend)

Yearly Trend:
               Sales   Profit
Order Year                   
2014       484247.50 49543.97
2015       470532.51 61618.60
2016       609205.60 81795.17
2017       733215.26 93439.27

Monthly Trend:
                Sales   Profit
Order Month                   
1            94924.84  9134.45
2            59751.25 10294.61
3           205005.49 28594.69
4           137762.13 11587.44
5           155028.81 22411.31
6           152718.68 21285.80
7           147238.10 13832.66
8           159044.06 21776.94
9           307649.95 36857.48
10          200322.98 31784.04
11          352461.07 35468.43
12          325293.50 43369.19


***Discount VS Profit Relationship***

In [41]:
# Check correlation between Discount and Profit
correlation = df['Discount'].corr(df['Profit'])
print("Correlation between Discount and Profit:", correlation)

# Group by Discount value and check average Profit
discount_profit = df.groupby('Discount')['Profit'].mean()
print("\nAverage Profit by Discount Level:")
print(discount_profit)

Correlation between Discount and Profit: -0.21948745637176834

Average Profit by Discount Level:
Discount
0.00     66.90
0.10     96.06
0.15     27.29
0.20     24.70
0.30    -45.68
0.32    -88.56
0.40   -111.93
0.45   -226.65
0.50   -310.70
0.60    -43.08
0.70    -95.87
0.80   -101.80
Name: Profit, dtype: float64


# **6** **Visulization**

***Sale Trend Over time***

In [42]:
import pandas as pd
import plotly.express as px
# Convert Order Date to proper date format
df['Order Date'] = pd.to_datetime(df['Order Date'])

# Extract Year-Month for grouping
df['Year Month'] = df['Order Date'].dt.to_period('M').astype(str)

# Group by Year Month and calculate total Sales
sales_trend = df.groupby('Year Month')['Sales'].sum().reset_index()

# Create Line Chart
fig = px.line(sales_trend, x='Year Month', y='Sales', title='Sales Trend Over Time')
fig.show()

***Profit By Catagory***

In [43]:
# Group by Category and calculate total Profit
category_profit = df.groupby('Category')['Profit'].sum().reset_index()

# Create Bar Chart
fig = px.bar(category_profit, x='Category', y='Profit', title='Profit By Category')
fig.show()

***Correlation Heatmap***

In [44]:
# Select numeric columns only
numeric_df = df[['Sales', 'Profit', 'Quantity', 'Discount']]

# Calculate correlation between columns
corr_matrix = numeric_df.corr()

# Create Heatmap
fig = px.imshow(corr_matrix, text_auto=True, color_continuous_scale='RdBu', title='Correlation Heatmap')
fig.show()

***Region Comparison chart***

In [45]:

# Group by Region and calculate total Sales and Profit
region_data = df.groupby('Region')[['Sales', 'Profit']].sum().reset_index()

# Create grouped Bar Chart
fig = px.bar(region_data, x='Region', y=['Sales', 'Profit'], barmode='group', title='Region Comparison - Sales vs Profit')
fig.show()

# 📊 Project 2 — Superstore Sales Analysis

## About this project

This is my second project in the Data Science course at Devsil — Superstore Sales Analysis. Honestly this one was way more detailed than the first project. I had to clean the data first, then do proper analysis, and also try to think like an actual business analyst instead of just running code.

The dataset is from Kaggle (Superstore Dataset), and it has sales, profit, and order data of a retail store from 2014 to 2017.

---

## Tools I used

- Python
- Pandas
- Plotly
- Jupyter Notebook

---

## What I actually did

I started by loading the CSV file and checking how the data looks — rows, columns, missing values, duplicates. Then I converted the date columns properly so I could work with year and month.

After that came the real analysis part:

- Grouped the data by Category, Sub-Category, Region and Segment to see where sales and profit are coming from
- Found out which areas are making money and which ones are losing money
- Created a new column called Profit Ratio to see how efficient each sale actually is (not just total numbers)
- Pulled out Year and Month from the order date to check trends over time
- Checked how discounts affect profit — this part was actually interesting
- Made a few charts using Plotly to see everything visually instead of just staring at tables

---

## Charts I made

- Sales trend over time (line chart)
- Profit by category (bar chart)
- Correlation heatmap
- Region comparison chart

Seeing it visually made a lot more sense than just reading numbers in a table. Like the seasonality thing — I wouldn't have noticed it just from raw numbers.

---

## Things I found out

- Technology makes the most profit, but Furniture is barely profitable even though its sales are almost as high as Technology
- Tables and Bookcases are actually losing money for the store
- West region is doing the best overall, Central is the weakest in terms of profit
- States like Texas, Ohio and Pennsylvania have decent sales but end up in loss — that surprised me
- Whenever discount goes above 30%, profit basically goes negative
- Sales spike every year in November-December, which makes sense because of holiday shopping

---

## What I learned from this project

This project made me actually think about the "why" behind the numbers, not just calculate them. It's one thing to know average discount is 15%, and a completely different thing to notice that high discounts are quietly killing the profit.

Still learning, still a beginner, but this one felt like a proper step up from project 1.

---

## Status: Done ✅

**Project:** Superstore Sales Analysis
**Next up:** more projects, more practice.